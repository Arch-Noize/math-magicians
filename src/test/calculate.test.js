import calculate from '../logic/calculate';

describe('Test the operates', () => {
  test('it should return the sum of numbers', () => {
    const obj = {
      total: 3,
      next: 3,
      operation: '+',
    };
    const btn = '=';

    expect(calculate(obj, btn)).toMatchSnapshot();
  });

  test('it should return the subtraction of numbers', () => {
    const obj = {
      total: 3,
      next: 3,
      operation: '-',
    };
    const btn = '=';

    expect(calculate(obj, btn)).toMatchSnapshot();
  });

  test('it should return the multiplication of numbers', () => {
    const obj = {
      total: 3,
      next: 3,
      operation: 'x',
    };
    const btn = '=';

    expect(calculate(obj, btn)).toMatchSnapshot();
  });

  test('it return the division of numbers', () => {
    const obj = {
      total: 3,
      next: 3,
      operation: 'รท',
    };
    const btn = '=';

    expect(calculate(obj, btn)).toMatchSnapshot();
  });

  test('it should return the modulus of numbers', () => {
    const obj = {
      total: 3,
      next: 3,
      operation: '%',
    };
    const btn = '=';

    expect(calculate(obj, btn)).toMatchSnapshot();
  });
});
