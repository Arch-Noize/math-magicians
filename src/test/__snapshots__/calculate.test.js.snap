// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the operates it return the division of numbers 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "1",
}
`;

exports[`Test the operates it should return the modulus of numbers 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "0",
}
`;

exports[`Test the operates it should return the multiplication of numbers 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "9",
}
`;

exports[`Test the operates it should return the subtraction of numbers 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "0",
}
`;

exports[`Test the operates it should return the sum of numbers 1`] = `
Object {
  "next": null,
  "operation": null,
  "total": "6",
}
`;
